### ISSUE TITLE
What is the Error Rate after Learning from Data 

### ISSUE THREAD

**Issue by TitusQuinctiusFlamininus** [tensorflow/haskell#197]
Hi. While running a typical example of supervised (regression learning), is it possible to see the **error rate** after all learning iterations are complete? It's true that we get a "w" and "b after we learn the data, but can we know how "accurate" those values are(before we apply it to our input)?
I realize there is some difficulty in printing tensors (or some part of the tensor type), so i am wondering how to see the error rate

**Comment by fkm3** [comment]
Using https://github.com/tensorflow/haskell/blob/master/tensorflow-ops/tests/RegressionTest.hs as a concrete example: It isn't obvious how to calculate the error rate for a regression problem because the `yHat` is rarely going to be exactly equal to the `y`. The loss function is the simplest way to measure the performance of the learned weights. You could get the final loss in the example by adding a line like

```haskell
(TF.Scalar loss') <- TF.run loss
```

to the end of `fit`.

**Comment by TitusQuinctiusFlamininus** [comment]
I have tried the above suggestion, that is: 

    `(ws', TF.Scalar b', TF.Scalar loss')        <- TF.run (TF.readValue ws, TF.readValue b, loss)
    return (w', b', loss')`

as well as (most likely an incorrect interpretation, since it dont think you should have 2 runs; but i tried it anyway):

```
(ws', TF.Scalar b')        <- TF.run (TF.readValue ws, TF.readValue b)
  (TF.Scalar loss')          <- TF.run loss
  return (w',b',loss')
```
and in both cases, i get the following error:

```
Unable to extract singleton from tensor of length 363
CallStack (from HasCallStack):
  error, called at src/TensorFlow/Types.hs:350:19 in tensorflow-0.1.0.2-1uses2Auj7IHV7CQxXVCdt:TensorFlow.Types
```

**Comment by TitusQuinctiusFlamininus** [comment]
The type of loss above turns out to be:
 :t loss 

`TF.Tensor TF.Build Float`

Excuse the newbie question, but how can i "see" this value (as a float) ? I looked up the docs, but i can't see a way.

**Comment by fkm3** [comment]
You can use `TF.run`. It's type signature is pretty general, but one specialization is `run :: TF.Tensor TF.Build Float -> TF.Session Float`:

```haskell
lossFloat <- TF.run loss
```

**Comment by fkm3** [comment]
(didn't mean to close, feel free to ask more questions)
